lib/common/auth.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+/**
+ * Module contains interface for AWS Authentication logic - signing http requests, events, chunks, etc...
+ *
+ * @packageDocumentation
+ * @module auth
+ * @preferred
+ */
+
+/**
+ * @internal
+ */
+export type StringLike = string | ArrayBuffer | DataView;
+
+/**
+ * AWS signing algorithm enumeration.
+ *
+ * @category Auth
+ */
+export enum AwsSigningAlgorithm {
```

Comment:
I'm not sure we want these to be shared.  We're never going to support these options in the browser so it seems weird to put them in the common definitions.